### PR TITLE
Fix metrics of the default Agroal data source

### LINKED ISSUE #7525: Metrics for default Agroal data source are broken
After https://github.com/quarkusio/quarkus/commit/0d64d8596242d1900b1f62334e2d89c6fbeaa3aa the metrics for Agroal default data source broke, because `AgroalCounter` and `AgroalGauge` expect `null` to be passed as the data source name when referring to the default DS. Now, instead of `null`, we pass them the name `<default>`, so the metric classes are unable to locate the correct data source.

This results in `java.lang.NullPointerException` appearing in the log every time metrics are queried, and metrics for the default data source don't appear in the output.

### PR DESCRIPTION
**Pull request by jmartisk**
Fixes #7525

### CODE REVIEW

**Review by geoand — APPROVED** [review]

File: extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java [hunk 1]
```diff
@@ -551,89 +551,91 @@ void registerMetrics(
             boolean metricsEnabledForThisDatasource = dataSourcesBuildTimeConfig.metricsEnabled &&
                     aggregatedDataSourceBuildTimeConfig.getJdbcConfig().enableMetrics.orElse(true);
             Tag tag = new Tag("datasource", DataSourceUtil.isDefault(dataSourceName) ? "default" : dataSourceName);
+            // Metric classes expect null as the name for the default DS
+            String dataSourceNameOrNull = DataSourceUtil.isDefault(dataSourceName) ? null : dataSourceName;
```
Comment (gsmet) [inline extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java]: Sorry I missed that but... could we adjust that, pass the name to the gauge and counter and use `DataSourceUtil.isDefault()` there?

That would be more consistent and avoid this sort of issues.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java (hunk 1 above)
Comment (jmartisk) [reply extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java]: Sounds like a good idea. Done

**Review by gsmet — APPROVED** [review]
Thanks!

**Comment by gsmet** [comment]
Force pushed a rebase.

**Comment by gsmet** [comment]
Thanks, good catch @jmartisk !

**Pull request merged**